dpe/src/commands/derive_context.rs

```diff
@@ -284,7 +288,39 @@ impl CommandExecution for DeriveContextCmd {
                 }
             }
         } else if self.creates_certificate() && self.exports_cdi() {
-            todo!("(clundin): Generate ECA certificiate / export random CDI")
+            cfg_if! {
+                if #[cfg(not(feature = "disable_export_cdi"))] {
+                    let mut exported_cdi_handle = [0; MAX_EXPORTED_CDI_SIZE];
+                    env.crypto
+                        .rand_bytes(&mut exported_cdi_handle)
+                        .map_err(DpeErrorCode::Crypto)?;
+                    let args = CreateDpeCertArgs {
+                        handle: &self.handle,
+                        locality,
+                        cdi_label: b"Exported CDI",
+                        key_label: b"Exported ECC",
+                        context: &exported_cdi_handle,
+                    };
+                    let mut cert = [0; MAX_CERT_SIZE];
+                    let CreateDpeCertResult { cert_size, .. } = create_exported_dpe_cert(
+                        &args,
+                        dpe,
+                        env,
+                        &mut cert,
+                    )?;
+
+                    Ok(Response::DeriveContextExportedCdi(DeriveContextExportedCdiResp {
+                        handle: ContextHandle::new_invalid(),
+                        parent_handle: dpe.contexts[parent_idx].handle,
```

Comment:
If export_cdi, then `retains_parent` must be false. So I think the parent handle should be `new_invalid`. We should see if we can share the logic from line 374 so it doesn't need to be duploicated.

Reply:
Moved the parent invalidation logic further up so it is shared with `export-cdi`.